web/src/features/moreCast2/forecastDraft.test.ts

```diff
@@ -0,0 +1,73 @@
+import { MorecastDraftForecast } from 'features/moreCast2/forecastDraft'
+import { DraftMorecast2Rows } from 'features/moreCast2/interfaces'
+import { buildValidActualRow, buildValidForecastRow } from 'features/moreCast2/rowFilters.test'
+import { DateTime } from 'luxon'
+import * as DateUtils from 'utils/date'
+
+const TEST_DATE = DateTime.fromISO('2024-01-01T00:00:00.000-08:00')
+
+describe('MorecastDraftForecast', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+  const localStorageMock: Storage = {
+    getItem: jest.fn(),
+    setItem: jest.fn(),
+    removeItem: jest.fn(),
+    length: 0,
+    clear: jest.fn(),
+    key: jest.fn()
+  }
+
+  const draftForecast = new MorecastDraftForecast(localStorageMock)
+
+  const mockRowData = [
+    buildValidForecastRow(111, TEST_DATE.plus({ days: 1 })),
+    buildValidForecastRow(111, TEST_DATE.plus({ days: 2 })),
+    buildValidForecastRow(222, TEST_DATE.plus({ days: 1 })),
+    buildValidForecastRow(222, TEST_DATE.plus({ days: 2 })),
+    buildValidActualRow(222, TEST_DATE.minus({ days: 1 })),
+    buildValidActualRow(222, TEST_DATE.minus({ days: 2 }))
+  ]
+
+  it('should only store forecast rows', () => {
+    jest.spyOn(DateUtils, 'getDateTimeNowPST').mockReturnValue(TEST_DATE)
```

Comment:
@conbrad Is this an ok solution? Still have some things to learn about ways to mock things...
It was actually `isBeforeToday` that was causing the issue, I changed the `forecastDraft` functions to not use `DateTime.now()` this morning like you suggested.
I just changed `isBeforeToday()` with this commit and mocked it with jest, since it seemed like I could do the mock the same way as sinon

Reply:
Yes! Nice!

Reply:
You'll actually want to clean up the spy after the test though -- with `sinon` this can be done with a sandbox, with `jest` you'll want something like this:

```
afterEach(() => {
  // restore the spy created with spyOn
  jest.restoreAllMocks();
});
```

If this is the only function that spies, you can just call `jest.restoreAllMocks();` at the end of the test.

Reply:
Ah I was using `clearAllMocks`, just spent a little more time reading about the differences. Thanks!